pkg/controlplane/apiserver/options/options.go

```diff
@@ -281,6 +283,9 @@ func (o *Options) Complete(alternateDNS []string, alternateIPs []net.IP) (Comple
 			delete(completed.APIEnablement.RuntimeConfig, key)
 		}
 	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CompatibilityVersion) {
+		completed.APIEnablement.CompatibilityVersion = ""
```

Comment:
Interesting, an alternative is to pass in the binary version.

Reply:
Wiring in the binary version is what I was expecting. It seems safer because it eliminates the need to do `if compat == ""` checks

Reply:
moved the feature gate check to validation and `NewResourceConfig`. 
`APIEnablement.EmulationVersion` is the binary version by default.